### ISSUE TITLE
你好，libtorch1.4好像用不了

### ISSUE THREAD

**Issue by bravestpeng** [walktree/libtorch-yolov3#43]
1.  142行 conv_options.with_bias(with_bias);只有这个 conv_options.bias(with_bias)，没有with_bias；
2.	at::TensorOptions options = torch::TensorOptions()
		.dtype(torch::kFloat32)
       .is_variable(true)，没有这个is_variable，我改成了at::TensorOptions options = torch::TensorOptions().dtype(torch::kFloat32);去掉了后面的.is_variable(true)；我改了之后，编译通过，GPU运行失败，出现错误的在yolo层，cpu上正常运行

**Comment by horsetif** [comment]
It is the version problem. Follow the https://pytorch.org/cppdocs/api/library_root.html. I change some code in Darknet.cpp.
line 133,
>     conv_options.bias(with_bias); // with_bias
line 143,
>     bn_options.track_running_stats(true);  //stateful
line 523,     
>     at::TensorOptions options= torch::TensorOptions().dtype(torch::kFloat32);
because in /usr/local/lib/python3.6/dist-packages/torch/include/ATen/core/TensorBody.h, it says:
>   C10_DEPRECATED_MESSAGE("Tensor.is_variable() is deprecated; everything is a variable now. (If you want to assert that variable has been appropriately handled already, use at::impl::variable_excluded_from_dispatch())")
  bool is_variable() const noexcept {
    return !at::impl::variable_excluded_from_dispatch();
  }

Then we can run the example without error and have the same results.
Futher, change all BatchNormXXX to BatchNorm2dXXX, so we can run without warning :
> BatchNormOptions -> BatchNorm2dOptions
BatchNorm -> BatchNorm2d
BatchNormImpl -> BatchNorm2dImpl

**Comment by bravestpeng** [comment]
Thank you for your help,I can not use
{        torch::Tensor tensor0 = torch::ones({ 169, 2 ,3}).to(torch::kCUDA);
         torch::Tensor tensor1 = torch::ones({ 169, 2 ,3}).to(torch::kCUDA);
          torch::Tensor  tensor2 = torch::cat({ tensor0 ,tensor1  }, 0);
},cuda is cuda::is_available in libtorch1.4 GPU
but I can use

{        torch::Tensor tensor0 = torch::ones({ 169, 2 ,3}).to(torch::kCPU);
         torch::Tensor tensor1 = torch::ones({ 169, 2 ,3}).to(torch::kCPU);
          torch::Tensor tensor2 = torch::cat({ tensor0 ,tensor1  }, 0);
}.
so torch::cat is different between GPU(cuda) an CPU，can you help me?

**Comment by horsetif** [comment]
My version of libtorch is 1.4.0, and use CUDA 10.0. I don't have this problem. Perhaps you can upload pytorch and CUDA.

**Comment by bravestpeng** [comment]
谢谢，我觉得这个代码还有一个地方应该改一下，darknet.cpp,360行，
if (layer_type == "net")		
	continue;
应该改成：
if (layer_type == "net")
{
	prev_filters = get_int_from_cfg(block, "channels", 0);
	continue;
}

**Comment by mheriyanto** [comment]
Wow awesome. Thank you @horsetif .
